pin/pin.go

```diff
@@ -221,30 +231,51 @@ func (p *pinner) RemovePinWithMode(key util.Key, mode PinMode) {
 func LoadPinner(d ds.ThreadSafeDatastore, dserv mdag.DAGService) (Pinner, error) {
 	p := new(pinner)
 
+	rootKeyI, err := d.Get(pinDatastoreKey)
```

Comment:
Today, migrations are manual. Even when they become automatic, things can still go wrong. We have users already, including (a) users who do not use the commandline, and (b) users who bundle IPFS to build other user-facing applications. They can't fix things manually. 

Last week we decided that we'll adopt a "dev branch" which merges into master, and we can merge this into that.